### PR TITLE
Pytorch Estimator support train on steps

### PR DESCRIPTION
**Pull request by jenniew**
Pytorch Estimator support train on max steps.

### CODE REVIEW

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py [hunk 1]
```diff
@@ -244,13 +244,18 @@ def train_epochs(self, data_creator, epochs=1, batch_size=32, profile=False,
         self.val_loader = val_loader
         self.call_hook(callbacks=callbacks, fn_name="before_run")
 
+        if max_steps is not None:
+            epochs = math.ceil(max_steps / len(self.train_loader))
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:248]: So `max_step` indicates a global step?

File: python/orca/src/bigdl/orca/learn/pytorch/pytorch_pyspark_estimator.py [hunk 2]
```diff
@@ -199,6 +199,7 @@ def _get_cluster_info(self, sc):
     def fit(self,
             data: Union['SparkXShards', 'SparkDataFrame', Callable[[Dict, int], 'DataLoader']],
             epochs: int=1,
+            max_steps: Optional[int]=None,
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/learn/pytorch/pytorch_pyspark_estimator.py:202]: Also in ray estimator?

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py [hunk 3]
```diff
@@ -361,17 +367,19 @@ def train_epoch(self, ...):
...
 
         self.call_hook(callbacks=callbacks, fn_name="on_lr_adjust")
 
         return metric_meters.summary(sync_stats=self.sync_stats,
                                      dist_backend=self.dist_backend)
 
-    def _train_loop(self, iterator, metric_meters, callbacks):
+    def _train_loop(self, iterator, metric_meters, callbacks, max_steps=None):
         for batch_idx, batch in enumerate(iterator):
+            if max_steps is not None and self.global_step >= max_steps:
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:381]: I prefer to integrate this into a EarlyStopping Callback after this merges? @hkvision We shall take it carefully to add any new arg in torchrunner, for they are really hard to be maintained.

Comment (jenniew) [reply python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:381]: earlystop is to stop with metric criteria. I don't think we need to integrate training with max steps to earlystop as max step means training only specific step without any condition. We may refer to pytorch lightning, max steps is a trainer options. EarlyStop is callback to stop when some condition met.

File: python/orca/src/bigdl/orca/learn/pytorch/pytorch_pyspark_estimator.py (hunk 2 above)
Comment (jenniew) [reply python/orca/src/bigdl/orca/learn/pytorch/pytorch_pyspark_estimator.py:202]: Yes, will also add to ray estimator.

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py (hunk 1 above)
Comment (jenniew) [reply python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:248]: global step is runner status, max steps is a training option for just training specific steps.

Comment (leonardozcm) [reply python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:248]: Yes, I mean the max_step is **compared** with the global step across all ranks?

File: python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py (hunk 3 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py:381]: You can keep this in this pr, and we may refactor this just like what we do in TqdmCallback? @hkvision  I think training only specific steps is also one kind of condition. What about we set up a flag like `runner.stop`, when any condition reaches we change `runner.stop` to True and runner checks it and exits in EarlyStop hooks.
